main/src/ca/uwaterloo/flix/language/phase/extra/CodeHinter.scala

```diff
@@ -42,12 +43,16 @@ object CodeHinter {
   /**
     * Computes code quality hints for the given enum `enum`.
     */
-  private def visitEnum(enum: TypedAst.Enum)(implicit root: Root): List[CodeHint] = {
+  private def visitEnum(enum: TypedAst.Enum)(implicit root: Root, index: Index): List[CodeHint] = {
```

Comment:
Let me propose something simple instead:

Since we have the enum, iterate through all its cases, then call index.usesOf(enum.sym, caze) and report a Deprecated/Experimental hint for each.

(This does not catch everything, in particular not types, but it is a good start!)

Reply:
Yes I was implementing something like this right now to try out, now see your comment ahah :)